test/benchmark.hs

```diff
@@ -47,17 +58,25 @@ main = do microbench "insNode into AVL tree" insNodeAVL
           microbench "nmap on AVL tree" nmapAVL
           microbench "nmap on PATRICIA tree" nmapPatricia
 
-          microbench "emap on AVL tree" emapAVL
+--          microbench "emap on AVL tree" emapAVL
           microbench "emap on PATRICIA tree" emapPatricia
 
 
 insNodeAVL :: Int -> AVL.UGr
 insNodeAVL = insNodes' empty
 
+buildFullPatricia :: Int -> Patricia.UGr
+buildFullPatricia = buildFull' empty 0
 
 insNodePatricia :: Int -> Patricia.UGr
 insNodePatricia = insNodes' empty
 
+{-# INLINE buildFull' #-}
+buildFull' :: DynGraph gr => gr () () -> Int -> Int -> gr () ()
+buildFull' g n limit
+  | n == limit = empty
+  | otherwise = ([((), k) | k <- [0..n-1]],n,(),[((),k) | k <- [0..n-1]]) & buildFull' g (n + 1) limit
+
```

Comment:
Actually, no, ignoring it won't work either. I have *no idea* how to make
my benchmark work with microbench.

On Tue, Aug 30, 2016 at 7:41 PM, David Feuer <david.feuer@gmail.com> wrote:

> No, the first argument is the graph to start with. But I'm using the Int
> parameter wrong, because we can't really use it right. I think probably the
> best we can do is ignore it.
>
> On Tue, Aug 30, 2016 at 7:37 PM, Ivan Lazar Miljenovic <
> notifications@github.com> wrote:
>
>> In test/benchmark.hs
>> <https://github.com/haskell/fgl/pull/46#discussion_r76899816>:
>>
>> >
>> >  insNodePatricia :: Int -> Patricia.UGr
>> >  insNodePatricia = insNodes' empty
>> >
>> > +{-# INLINE buildFull' #-}
>> > +buildFull' :: DynGraph gr => gr () () -> Int -> Int -> gr () ()
>> > +buildFull' g n limit
>> > +  | n == limit = empty
>> > +  | otherwise = ([((), k) | k <- [0..n-1]],n,(),[((),k) | k <- [0..n-1]]) & buildFull' g (n + 1) limit
>> > +
>>
>> I assume the first argument is just for a proxy?
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/haskell/fgl/pull/46/files/95215cb25d346d910f7fb11dcc4dddd3e316a95d#r76899816>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/ABzi_cbvK3lbHfN1UtAIiNVjs4pKJIXvks5qlL6jgaJpZM4Jw2_G>
>> .
>>
>
>